consensus/spos/bls/subroundSignature_test.go

```diff
@@ -872,109 +907,196 @@ func TestSubroundSignature_SendSignature(t *testing.T) {
-	container.SetBroadcastMessenger(&mock.BroadcastMessengerMock{
-		BroadcastConsensusMessageCalled: func(message *consensus.Message) error {
-			mutex.Lock()
-			signaturesBroadcast[string(message.PubKey)]++
-			mutex.Unlock()
-			return nil
-		},
 	})
 
-	sr.SetSelfPubKey("OTHER")
+	t.Run("should fail", func(t *testing.T) {
+		t.Parallel()
+		container := mock.InitConsensusCore()
+		enableEpochsHandler := &enableEpochsHandlerMock.EnableEpochsHandlerStub{
+			IsFlagEnabledInEpochCalled: func(flag core.EnableEpochFlag, epoch uint32) bool {
+				return flag == common.EquivalentMessagesFlag
+			},
+		}
+		container.SetEnableEpochsHandler(enableEpochsHandler)
 
-	r := srSignature.DoSignatureJobForManagedKeys()
-	assert.True(t, r)
+		signingHandler := &consensusMocks.SigningHandlerStub{
+			CreateSignatureShareForPublicKeyCalled: func(msg []byte, index uint16, epoch uint32, publicKeyBytes []byte) ([]byte, error) {
+				return []byte("SIG"), nil
+			},
+		}
+		container.SetSigningHandler(signingHandler)
+		consensusState := initConsensusStateWithKeysHandler(
+			&testscommon.KeysHandlerStub{
+				IsKeyManagedByCurrentNodeCalled: func(pkBytes []byte) bool {
+					return true
+				},
+			},
+		)
+		ch := make(chan bool, 1)
 
-	for _, pk := range sr.ConsensusGroup() {
-		isJobDone, err := sr.JobDone(pk, bls.SrSignature)
-		assert.NoError(t, err)
-		assert.True(t, isJobDone)
-	}
+		sr, _ := spos.NewSubround(
+			bls.SrBlock,
+			bls.SrSignature,
+			bls.SrEndRound,
+			int64(70*roundTimeDuration/100),
+			int64(85*roundTimeDuration/100),
+			"(SIGNATURE)",
+			consensusState,
+			ch,
+			executeStoredMessages,
+			container,
+			chainID,
+			currentPid,
+			&statusHandler.AppStatusHandlerStub{},
+		)
 
-	expectedMap := map[string]struct{}{
-		"A": {},
-		"B": {},
-		"C": {},
-		"D": {},
-		"E": {},
-		"F": {},
-		"G": {},
-		"H": {},
-		"I": {},
-	}
-	assert.Equal(t, expectedMap, signatureSentForPks)
-
-	expectedBroadcastMap := map[string]int{
-		"B": 1,
-		"C": 1,
-		"D": 1,
-		"E": 1,
-		"F": 1,
-		"G": 1,
-		"H": 1,
-		"I": 1,
-	}
-	assert.Equal(t, expectedBroadcastMap, signaturesBroadcast)
+		signatureSentForPks := make(map[string]struct{})
+		mutex := sync.Mutex{}
+		srSignature, _ := bls.NewSubroundSignature(
+			sr,
+			&statusHandler.AppStatusHandlerStub{},
+			&testscommon.SentSignatureTrackerStub{
+				SignatureSentCalled: func(pkBytes []byte) {
+					mutex.Lock()
+					signatureSentForPks[string(pkBytes)] = struct{}{}
+					mutex.Unlock()
+				},
+			},
+			&mock.SposWorkerMock{},
+			&mock2.ThrottlerStub{
+				CanProcessCalled: func() bool {
+					return false
+				},
+			},
+		)
+
+		sr.Header = &block.Header{}
+		signaturesBroadcast := make(map[string]int)
+		container.SetBroadcastMessenger(&mock.BroadcastMessengerMock{
+			BroadcastConsensusMessageCalled: func(message *consensus.Message) error {
+				mutex.Lock()
+				signaturesBroadcast[string(message.PubKey)]++
+				mutex.Unlock()
+				return nil
+			},
+		})
```

Comment:
there is no need to set broadcast messenger, since it is not being used in this one, so the test can be a bit simpler, or at least just to set an empty mock broadcast messenger

Reply:
modified